test/test_typeregistry.cpp

```diff
@@ -0,0 +1,69 @@
+#define BOOST_TEST_MAIN
+#define BOOST_TEST_MODULE "test_typeregistry"
+#define BOOST_AUTO_TEST_MAIN
+
+#include <boost/bind.hpp>
+#include <boost/test/unit_test.hpp>
+#include <boost/test/execution_monitor.hpp>  
+
+#include <orocos_cpp/TypeRegistry.hpp>
+
+using namespace orocos_cpp;
+
+
+class TypeRegistryTest: public TypeRegistry
+{
+public:
+    TypeRegistryTest():TypeRegistry(){}
+
+    void loadStatesFromCustomPath(const std::string &path)
+    {
+        loadStateToIDMapping(path);
+    }
+
+    void loadTypesFromCustomPath(const std::string &path, const std::string &typekitName)
+    {
+        loadTypeToTypekitMapping(path, typekitName);
+    }
+};
+
+struct Fixture
+{
+    Fixture()
+    {
+        type_registries.loadStatesFromCustomPath("../../test/testfile.tlb");
+        type_registries.loadTypesFromCustomPath("../../test/testfile.typelist", "auv_control");
```

Comment:
The test files must be installed as well. The path of the build folder isn't fixed.

One solution would be to add
```
configure_file(${CMAKE_CURRENT_SOURCE_DIR}/testfile.tlb
               ${CMAKE_CURRENT_BINARY_DIR}/testfile.tlb COPYONLY)
```
and assume that the test files are in the same folder than the test executable.